testing/sev_snp_hello_world_kernel/src/main.rs

```diff
@@ -17,16 +17,30 @@
 #![no_std]
 #![no_main]
 
-use core::panic::PanicInfo;
+use core::{mem::MaybeUninit, panic::PanicInfo};
+use sev_guest::{cpuid::CpuidPage, secrets::SecretsPage};
 use x86_64::instructions::{hlt, port::PortWriteOnly};
 
 mod asm;
 mod serial;
 
+#[link_section = ".cpuid"]
+static CPUID: MaybeUninit<CpuidPage> = MaybeUninit::uninit();
+
+#[link_section = ".secrets"]
+static SECRETS: MaybeUninit<SecretsPage> = MaybeUninit::uninit();
+
 #[no_mangle]
 pub extern "C" fn rust64_start() -> ! {
     serial::init_logging();
     log::info!("Hello World!");
+
+    #[allow(dead_code)]
+    let _cpuid: &CpuidPage = unsafe { CPUID.assume_init_ref() };
```

Comment:
Why does it need to be assigned to a local variable? Does it get optimized away otherwise?

Reply:
Yep, Rust optimized away the whole static, much to my surprise -- even if I tagged it with `#[used]`.

However, I've added some statements logging the contents of these pages (as we're interested what's in them anyway), so these are no longer unused.